lib/ansible/modules/remote_management/oneview/oneview_datacenter_facts.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/python
+# Copyright (c) 2016-2017 Hewlett Packard Enterprise Development LP
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: oneview_datacenter_facts
+short_description: Retrieve facts about the OneView Data Centers
+description:
+    - Retrieve facts about the OneView Data Centers.
+version_added: "2.5"
+requirements:
+    - "hpOneView >= 2.0.1"
+author:
+    - Alex Monteiro (@aalexmonteiro)
+    - Madhav Bharadwaj (@madhav-bharadwaj)
+    - Priyanka Sood (@soodpr)
+    - Ricardo Galeno (@ricardogpsf)
+options:
+    name:
+      description:
+        - Data Center name.
+    options:
+      description:
+        - "Retrieve additional facts. Options available: 'visualContent'."
+
+extends_documentation_fragment:
+    - oneview
+    - oneview.factsparams
+'''
+
+EXAMPLES = '''
+- name: Gather facts about all Data Centers
+  oneview_datacenter_facts:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+  no_log: true
+  delegate_to: localhost
+- debug: var=datacenters
+
+- name: Gather paginated, filtered and sorted facts about Data Centers
+  oneview_datacenter_facts:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    params:
+      start: 0
+      count: 3
+      sort: 'name:descending'
+      filter: 'state=Unmanaged'
+  no_log: true
+- debug: var=datacenters
+
+- name: Gather facts about a Data Center by name
+  oneview_datacenter_facts:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    name: "My Data Center"
+  no_log: true
+  delegate_to: localhost
+- debug: var=datacenters
+
+- name: Gather facts about the Data Center Visual Content
+  oneview_datacenter_facts:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    name: "My Data Center"
+    options:
+      - visualContent
+  no_log: true
+  delegate_to: localhost
+- debug: var=datacenters
+- debug: var=datacenter_visual_content
+'''
+
+RETURN = '''
+datacenters:
+    description: Has all the OneView facts about the Data Centers.
+    returned: Always, but can be null.
+    type: dict
+
+datacenter_visual_content:
+    description: Has facts about the Data Center Visual Content.
+    returned: When requested, but can be null.
+    type: dict
+'''
+
+from ansible.module_utils.oneview import OneViewModuleBase
+
+
+class DatacenterFactsModule(OneViewModuleBase):
+    argument_spec = dict(
+        name=dict(type='str'),
+        options=dict(type='list'),
+        params=dict(type='dict')
+    )
```

Comment:
I *think* we want `supports_check_mode=True`

Reply:
At this moment, by default, the check_mode is disabled for all oneview modules directly in the [oneview.py module_utils](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/oneview.py#L228). 
We do plan on supporting it, but since it would require many changes on all of the modules, that is in a "to do" list at the moment.

Reply:
Sure, that sounds reasonable.